File: leetcode_problems/cpp/lc1863_xor_of_subsets.cpp
#include <vector>
using std::vector;

class Solution
{
public:
    void backtrack(vector<int> &nums, int i, int xr, int &sum)
    {
        sum += xr;
        for (int j = i; j < nums.size(); j++)
        {
            xr ^= nums[j];
            backtrack(nums, j + 1, xr, sum);
            xr ^= nums[j];
        }
    }

    int subsetXORSum(vector<int> &nums)
    {
        int sum = 0;
        int xr = 0;
        backtrack(nums, 0, xr, sum);
        return sum;
    }
};

File: leetcode_problems/typescript/src/lc1863_xor_of_subsets.ts
function subsetXORSum(nums: number[]): number {
  function backtrack(i: number) {
    sum += xor;
    for (let j = i; j < nums.length; j++) {
      xor ^= nums[j];
      backtrack(j + 1);
      xor ^= nums[j];
    }
  }

  let sum = 0;
  let xor = 0;
  backtrack(0);
  return sum;
}
